gh-123961: remove global variable `ModDict` in `_cursesmodule.c`

picnixz opened the pull request:
<!-- gh-issue-number: gh-123961 -->
* Issue: gh-123961
<!-- /gh-issue-number -->

picnixz committed:
remove global variable `ModDict`

vstinner commented on Modules/_cursesmodule.c:
```diff
@@ -3977,10 +3979,11 @@ _curses_qiflush_impl(PyObject *module, int flag)
 
 /* Internal helper used for updating curses.LINES, curses.COLS, _curses.LINES
- * and _curses.COLS */
+ * and _curses.COLS. */
```
Put `Returns 1 on success and 0 on failure.` here.

picnixz replied:
Would it actually be better that I move the "Internal helper" comment inside the `if defined`-guarded block? That way, we wouldn't have an orphan comment on systems that don't support the feature.

vstinner reviewed (approved):
LGTM, but a comment is put in a weird place, just move it.

picnixz committed:
Avoid orphan comment for `update_lines_cols`.

This moves the comment inside the `#if`-guarded block so that IDEs may choose not to render that comment on builds that do not support `update_lines_cols`.
